momento-sdk/src/main/java/momento/sdk/messages/CacheSetIfNotExistsResponse.java

```diff
@@ -0,0 +1,109 @@
+package momento.sdk.messages;
+
+import com.google.protobuf.ByteString;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import momento.sdk.exceptions.SdkException;
+import momento.sdk.internal.StringHelpers;
+
+/**
+ * Parent response type for a cache setIfNotExists request. The response object is resolved to a
+ * type-safe object of one of the following subtypes:
+ *
+ * <p>- {Stored} - {NotStored} - {Error}
+ *
+ * <p>`instanceOf` type guards are used to operate on the appropriate subtype @Example For example:
+ *
+ * <p>``` if (response instanceof CacheSetIfNotExists.Error) { // Handle error as appropriate. The
+ * compiler will smart-cast `response` to type `CacheSetIfNotExists.Error` // in this block, so that
```

Comment:
Java doesn't auto cast variables in if statements like kotlin does. Java 17 can almost do this. I would remove this example since the other response classes don't have one and we can add one later if we update the documentation.